packages/babel-traverse/src/path/inference/index.ts

```diff
@@ -81,11 +81,14 @@ export function _getTypeAnnotation(this: NodePath): any {
   typeAnnotationInferringNodes.add(node);
 
   try {
-    let inferer = inferers[node.type];
+    let inferer =
+      // @ts-expect-error inferers do not cover all AST types
+      inferers[node.type];
```

Comment:
The `inferers` could be refined to `Record<node["type"], Inferer>` but then we have to convert the namespace exports in `inferers` to a plain object export.